bokeh/charts/chart.py

```diff
@@ -322,46 +298,71 @@ def make_grid(self, dimension, ticker):
 
         return grid
 
-    def show(self):
-        """Main show function.
-
-        It shows the plot in file, server and notebook outputs.
-        """
-        # Add to document and session
-        if self.server:
-            if self.server is True:
-                self._servername = "untitled_chart"
-            else:
-                self._servername = self.server
-
-            self._session.use_doc(self._servername)
-            self._session.load_document(self._doc)
-
-        if not self._doc._current_plot == self:
-            self._doc._current_plot = self
-            self._doc.add_root(self)
 
-        if self.filename:
-            if self.filename is True:
-                filename = "untitled"
-            else:
-                filename = self.filename
-
-            with open(filename, "w") as f:
-                f.write(file_html(self._doc, INLINE, self.title))
-            print("Wrote %s" % filename)
-            view(filename)
-        elif self.filename is False and \
-                        self.server is False and \
-                        self.notebook is False:
-            print("You must provide a filename (filename='foo.html' or"
-                  " .filename('foo.html')) to save your plot.")
-
-        if self.server:
-            self.session.store_document(self._doc)
-            link = self._session.object_link(self._doc.context)
-            view(link)
-
-        if self.notebook:
-            from bokeh.embed import notebook_div
-            publish_display_data({'text/html': notebook_div(self)})
+    @property
+    def filename(self):
+        warnings.warn(
+            """
+            Chart property 'filename' will be deprecated in Bokeh
+            0.12.0. Use 'bokeh.io.output_file' instead.
```

Comment:
Better message is to say it "was deprecated in `0.11` and will be removed in the future". It's not certain there will be a `0.12`